### PR TITLE
fix lightningd config not receiving lightning environment variables from docker-compose

### PR DESCRIPTION
**Pull request by eoliveros**
- [x] create a static config file for the lightningd container

### CODE REVIEW

File: lightningd/config [hunk 1]
```diff
@@ -0,0 +1,7 @@
+bitcoin-datadir=/etc/bitcoin
+bitcoin-rpcconnect=bitcoind
+
+bind-addr=0.0.0.0:9735
+network=testnet
```
Comment (djpnewton) [inline lightningd/config:5]: this will not work for mainnet

**Review by djpnewton — COMMENTED** [review]
we already have config settings for these in the docker compose file:
```
    environment:
      LIGHTNINGD_NETWORK: ${BITCOIN_NETWORK:?BITCOIN_NETWORK not set}
      LIGHTNINGD_CHAIN: btc
      LIGHTNINGD_ALIAS: benode
      LIGHTNINGD_PORT: 9735
      LIGHTNINGD_REPLICA: ${LIGHTNINGD_REPLICA}
      LIGHTNINGD_OPT: |
        bitcoin-datadir=/etc/bitcoin
        bitcoin-rpcconnect=bitcoind
```
if they are not working (could be, because they came from when we used the btcpayserver image) then they should be removed in favor of this config file

File: lightningd/config (hunk 1 above)
Comment (eoliveros) [reply lightningd/config:5]: changing this so that my script would be run and grab the variable from the env and create it.

Comment (djpnewton) [reply lightningd/config:5]: maybe easier to use the python package jinja2-cli (https://github.com/mattrobenolt/jinja2-cli/blob/d66d6195b2fa4bb3d161e8dbdd2c3e9a8e8e1ecf/samples/environ.jinja2) which can template files using environment variables

Comment (eoliveros) [reply lightningd/config:5]: replacing my script to be the one above.

**Comment by djpnewton** [comment]
the config generator file can be simplified I think and also made to not override the defaults if an environment var is not supplied

eg:

```
# clear config file
cat <<-EOF > "$LIGHTNINGD_DATA/config"
# set arbitrary lightning options
if [ -n "$LIGHTNINGD_OPT" ]; then
  echo "$LIGHTNINGD_OPT" >> "$LIGHTNINGD_DATA/config"
fi
# set network option
if [ -n "$LIGHTNINGD_NETWORK" ]; then
  echo "network=$LIGHTNINGD_NETWORK" >> "$LIGHTNINGD_DATA/config"
fi
etc..
```

**Review by djpnewton — CHANGES_REQUESTED** [review]

File: lightningd/entrypoint-shim.sh [hunk 2]
```diff
@@ -7,4 +7,7 @@ if [ -n "$LIGHTNINGD_REPLICA" ]; then
   wallet="$wallet:/root/.lightning_replica/lightningd.sqlite3"
 fi
 
+# using jinja2cli, generate the config
+jinja2 /config.j2 > /root/.lightning/config
```
[the remark this answers is not in the document]
Comment (eoliveros) [reply lightningd/entrypoint-shim.sh]: change to something similar

File: lightningd/Dockerfile [hunk 3]
```diff
@@ -7,6 +7,8 @@ COPY ./requirements.txt /tmp/requirements.txt
 RUN pip3 install --upgrade pip
 RUN pip3 install -r /tmp/requirements.txt
 COPY ./rebalance.py /etc/bundledplugins/
+# copy config.j2 file to the root directory
```
[the remark this answers is not in the document]
Comment (eoliveros) [reply lightningd/Dockerfile:10]: removed the line

File: lightningd/config.j2 [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{{environ('LIGHTNINGD_OPT')}}
+bind-addr=0.0.0.0:{{environ('LIGHTNINGD_PORT')}}
```
[the remark this answers is not in the document]
Comment (eoliveros) [reply lightningd/config.j2]: added some defaults if the var in the env is not present.  doing test to make sure lightningd is still running.

[the remark this answers is not in the document]
Comment (eoliveros) [reply lightningd/config.j2]: lightningd is still running when i removed the LIGHTNINGD_OPT, so it appears to be working.

**Review by djpnewton — CHANGES_REQUESTED** [review]

**Comment by eoliveros** [comment]
deleted the commented line.

**Comment by djpnewton** [comment]
`LIGHTNINGD_CHAIN` is not in the config file

**Comment by eoliveros** [comment]
> `LIGHTNINGD_CHAIN` is not in the config file

checking the following documentation:
https://github.com/ElementsProject/lightning/blob/master/doc/lightningd-config.5.md

No idea where LIGHTNINGD_CHAIN is suppose to fit.

**Comment by djpnewton** [comment]
> > `LIGHTNINGD_CHAIN` is not in the config file
> 
> checking the following documentation: https://github.com/ElementsProject/lightning/blob/master/doc/lightningd-config.5.md
> 
> No idea where LIGHTNINGD_CHAIN is suppose to fit.

ok maybe its from the btcpayserver image, get rid of it from the docker-compose file then if it does not do anything

**Comment by eoliveros** [comment]
> > > `LIGHTNINGD_CHAIN` is not in the config file
> > 
> > 
> > checking the following documentation: https://github.com/ElementsProject/lightning/blob/master/doc/lightningd-config.5.md
> > No idea where LIGHTNINGD_CHAIN is suppose to fit.
> 
> ok maybe its from the btcpayserver image, get rid of it from the docker-compose file then if it does not do anything

deleted the line with LIGHTNINGD_CHAIN and build the image and run. not seeing any errors so far will commit and push this.

**Comment by djpnewton** [comment]
nice

**Pull request merged**